File: examples/python/connect_minio.py
# Hippius S3 — Connect with MinIO SDK
# Docs: /storage/s3/python
#
# Create a .env file (see .env.example):
#   HIPPIUS_ACCESS_KEY=hip_...
#   HIPPIUS_SECRET_KEY=...

import os
from dotenv import load_dotenv
from minio import Minio

load_dotenv()

client = Minio(
    "s3.hippius.com",
    access_key=os.environ["HIPPIUS_ACCESS_KEY"],
    secret_key=os.environ["HIPPIUS_SECRET_KEY"],
    secure=True,
    region="decentralized",
)

buckets = client.list_buckets()
print(f"Connected. {len(buckets)} bucket(s) found.")

File: examples/javascript/connect-minio.mjs
// Hippius S3 — Connect with MinIO SDK
// Docs: /storage/s3/javascript
//
// Create a .env file (see .env.example):
//   HIPPIUS_ACCESS_KEY=hip_...
//   HIPPIUS_SECRET_KEY=...

import "dotenv/config";
import { Client } from "minio";

const client = new Client({
  endPoint: "s3.hippius.com",
  port: 443,
  useSSL: true,
  accessKey: process.env.HIPPIUS_ACCESS_KEY,
  secretKey: process.env.HIPPIUS_SECRET_KEY,
  region: "decentralized",
});

const buckets = await client.listBuckets();
console.log(`Connected. ${buckets.length} bucket(s) found.`);
